### DISCUSSION TITLE
How can I use a different starship.toml based on directory?

### DISCUSSION THREAD

**Discussion by metaminimalist** [starship/starship#3229] [Q&A] [answered]
I set up starship, and it was good… and we lived happily ever after… NOT!

It worked out, until I entered one of my FUSE-mounted ftps-accounts, which is freakish slow normally (like 33.6k modem slow), and everything froze… froze…froze… until starship was ready looking und testing for git, yaml, toml files on the slow remote side to show its best of class shell prompt.

and this incident will be repeated with every commend, change directory or other shell interaction.

I closed several terminal windows with xkill today, due to anger ;)

How can I change to a minimal starship prompt in my FUSE-mounts directory, but have the fully riced-out starship experience anywhere else?

**Comment by davidkna** [comment, marked as answer]
starship has a `scan_timeout` option that is supposed to prevent this, but it's imperfect and it doesn't affect the initial opening of the directory. You can use the `STARSHIP_CONFIG` variable to use different config file and something like direnv to automatically set the variable.

**Reply by metaminimalist** [reply]
It took a bit of fiddeling (remote site does not accept dotfiles to be written) so I now have a directory containing a dotenv file and a subdirectory on which I mount the remote ftps.
_"it's not pretty, but it works"_ **Thank you!**
